vm/systemSmartContracts/delegation.go

```diff
@@ -1322,6 +1332,31 @@ func (d *delegation) reDelegateRewards(args *vmcommon.ContractCallInput) vmcommo
 		args.RecipientAddr, delegateValue, delegateValue, false, dConfig.CheckCapOnReDelegateRewards)
 }
 
+func (d *delegation) checkNotToReDelegateDust(delegator *DelegatorData) error {
+	if !d.flagReDelegateDustCheck.IsSet() {
+		return nil
+	}
+
+	delegationManagement, err := getDelegationManagement(d.eei, d.marshalizer, d.delegationMgrSCAddress)
+	if err != nil {
+		return err
+	}
+
+	fund, err := d.getFund(delegator.ActiveFund)
```

Comment:
here you may have the scenario that the user does not have any active fund - thus this get will return error - the user might still have more than 1 eGLD as rewards.

Reply:
Does the active fund gets erased even in the case the delegator undelegated everything but did not claimed its rewards?

Reply:
I think this will return the error on L1289 because the computeAndUpdateRewards try to fetch the active found object. So it is ok to have the code as it is

Reply:
fixing

Reply:
yes. active fund is deleted.